### ISSUE TITLE
Server Specs

### ISSUE THREAD

**Issue by jeffdavidgordon** [wet-boew/wet-boew-drupal#448]
Hi all,

Just wondering what kind of server hardware specifications are needed to optimally run WET-Drupal?   What is everyone else using?

We have about 100,000 pages with a peak of 130,000 page views per day.

We'll have a separate publishing server that will need to handle 50-100 publishers.

We're hoping to leverage the SSC cluster option being proposed, but until then we need a backup option to reach a July release date.   I'm willing to help wherever I can with the SSC cluster option, though I'm still quite new to Vagrant and Chef.    I'm used to doing server configuration the old fashioned way of doing it all manually.

Any help is kindly appreciated!

**Comment by StephenOTT** [comment]
You mention you want to have a separate publishing server.  You want to publish to that server and then replicate to the public prob server?

**Comment by thriuin** [comment]
Infrastructure requirements will vary on how you configure your system. I think the general assumption is that a dedicated Linux server offers the best performance. Assuming your web server (and web caching software) are on one server and your database is on another, then you probably want at least a 2-core CPU on your servers and at least 2 GB memory - possibly high as 8 GB if you are running Varnish on the webserver. MySQL can take advantage of multi-core servers (Percona and other derivatives even more so) so you may want to have a 4 core for the database. Drupal and MySQL by themselves don't take up much disk space, so the about of disk space you need available will vary with the size of your content.

**Comment by sylus** [comment]
A very rough guideline each in order of performance.

1) Web Server: NGINX, Apache, IIS7
2) Database: MySQL (Percona, Oracle Supported 5.6, MariaDB), PostGres
3) Reverse Proxy Cache: Varnish (Preconfigured Default.VCL)
4) DB Optmizations: Memcache

As for the server itself @thriuin is bang on.

**Comment by RobJohnston** [comment]
Since the OP also asked what everybody else was running, I'll add our (non-optimal) configuration...  IIS6 running PHP on Windows via FastCGI against MS SQL Server.  

We have one non-WET instance of Drupal in production on a beefy machine getting lots of hits and in the next few months hope to roll out some WET instances on sites that receive little traffic.

I often find that I have to dig around to find patches to get MS SQL Server to work correctly with some modules, but when I do it doesn't given any more problems.

**Comment by sylus** [comment]
Interesting note to mention @RobJohnston is that IIS7 + FastCGI outperforms Apache by about 25% :) So definitely another route to go if people want though not Linux based so not my recommendation (personal bias) Though NGINX vastly outperforms both sadly can't see departments using it.

**Comment by jeffdavidgordon** [comment]
Thanks so much for all of your feedback!   Sorry I didn't get back to this thread sooner.

@StephenOtt About the publishing server, yes that's what we want to do.   We'd like to keep the pre-published data seperate from the published data for security reasons and limit the amount of writing to Production to only very defined mechanisms (such as a deployment module).

**Comment by jeffdavidgordon** [comment]
So far for the OS/software stack, we've running (or plan to run):
1) Varnish
2) Apache 2.2
3) PHP-FPM 
4) APC
5) Memcached
6) MySQL 5
7) RHEL 5.x

For the hardware, we've got a VM provisioned to 1 CPU duo-core, 3GB RAM.    

Still working on tuning the settings og all of the parts as I've just got them all installed.   Without most of the above stack, we were getting hammered on the CPU usage, but it's improved quite a bit even before tuning.
